src/auth/auth.controller.ts

```diff
@@ -36,22 +34,14 @@ export class AuthController {
       httpOnly: true,
       expires: TWO_HOURS_FROM_NOW_DATE,
     });
-    return { message: 'Login successful' };
+    return { message: 'Login successful', access_token: access_token };
```

Comment:
Why do we need to return the access token in the login method? You've already set the cookie.

Also please move all of the business logic to the auth service. Setting the cookie should be there, not in the controller.

Reply:
It was like that in the past, I think it is unuseful to pass the `response` to the auth service only to return the cookie, also only the controller should "talk" with the frontend so passing the response object is wrong in my opinion

Reply:
about the access token, I did it for the websockets purpse, I think it will be useful for the frontend to add the access token to the websockets chat, if it makes more sense I could remove it but I want it to be easier to the frontend to add the cookie to the `WS` request

Reply:
But there's no need to send the access_token since you've already set the cookie to frontend. They can already parse the cookie in frontend and adjust a request accordingly to the websockets requests.

Also you have business logic here, this should be within the service.

Reply:
ok mate I agree on the first thing but not on the second, lets talk about it

Reply:
It it pointless to pass the request object just for doing `res.cookie` , besides the service should not talk with the frontend, only the controller should